File: CS201/Assignments/PythonJavaAll/Lists.java
import java.util.List;
import java.util.ArrayList;

/**
 * Lists.java
 * Jeff Ondich, 2014-01-05
 * Modified: Anna Rafferty, 2016-08-24
 * Modified: Layla Oesper, 2017-08-30
 *
 * Demonstrates a few operations with Java List interface and ArrayList objects.
 *
 * This is the Java half of a pair of parallel examples in Python and Java.
 * See lists.py.
 */
public class Lists {
    public static void main(String[] args) {
        // Instantiate an empty list.
        List<String> animalList = new ArrayList<String>();
        
        // Add some items to the list.
        animalList.add("goat");
        animalList.add("moose");
        animalList.add("emu");
        animalList.add("kudu");
        animalList.add("coatimundi");
        
        // Insert an item at index 2 (*third* position, after indices 0 and 1).
        animalList.add(2, "gnat");
        
        // Iterate over the items in the list.
        System.out.println("Animals iterated using a for (:) loop");
        for (String animal : animalList) {
            System.out.println(animal);
        }
        
        System.out.println("");
        System.out.println("Animals iterated using an index");
        for (int index = 0; index < animalList.size(); index++) {
            System.out.println(animalList.get(index));
        }
    }
}

File: CS201/Assignments/PythonJavaAll/lists.py
'''
   lists.py
   Jeff Ondich, 2013-01-05
   Anna Rafferty, 2016-08-24 (conversion to python3)

   Demonstrates a few operations with lists.
   
   Intended as the Python half of parallel examples in Python and
   Java. See Lists.java.
'''

# Initialize
animal_list = ['goat', 'moose', 'emu', 'kudu']

# Add an item
animal_list.append('coatimundi')

# Insert an item in third position
animal_list.insert(2, 'gnat')

# Iterate over the items in the list using a for loop
print('Animals iterated using a for/in loop')
for animal in animal_list:
    print(animal)

print()
print('Animals iterated using an index')
for k in range(len(animal_list)):
    print(animal_list[k])